### PR TITLE
[#7] [functional-test] Have the hook server control the review labels

### PR DESCRIPTION
**Pull request by adiroiban**
Scope
======

This PR is here to implement label management for the review process.

The idea is that we will continue to use our normal review workflows and the GitHub hooks server will auto-set the labels so that we can filter the reviews

Changes
======

When a review is requested, all reviewers are set as assignees for the PR.
In this way, we know which reviewers are required, and which are optional... and which of the required reviewers have not yet reviewed the PR. The needs-review label is set and all other labels are removed.

When changes are requested, the needs-review  (or needs-merge) label is removed and needs-changes is set. The PR requester is set as assignee.

When changes are approved

This adds git-secret to manage the private testing data.

--------

actions can be done as comments and as dedicated review actions.

needs-review can only be requested as a comment.

This is used for testing the review process.

This is used by the automated tests for the functional tests. Don't close it and make sure you unsubscribe from it.

When a review is requested, the reviewers are set as assignees.
When a review is done, they are removed.

The labels are also updated.

In this way, it should be easier to generate a report.

How to test
=========

reviewers: @hcs0 @lgheorghiu @dumol 

manually set various labels and see how they are updated after an action.

Trigger various actions and see how the labels/assignees are auto updated.

### CODE REVIEW

**Comment by adiroiban** [comment]
needs-review

**Comment by adiroiban** [comment]
still testing... sorry for the noise

needs-review

**Comment by adiroiban** [comment]
needs-review

**Comment by adiroiban** [comment]
needs-review

**Comment by adiroiban** [comment]
needs-review

**Comment by adiroiban** [comment]
needs-review

**Comment by adiroiban** [comment]
needs-review

**Comment by adiroiban** [comment]
needs-review

**Comment by adiroiban** [comment]
needs-changes

**Comment by adiroiban** [comment]
needs-changes

**Comment by adiroiban** [comment]
needs-changes

**Comment by adiroiban** [comment]
needs-changes

**Comment by adiroiban** [comment]
needs-changes

**Comment by adiroiban** [comment]
needs-changes

**Comment by adiroiban** [comment]
needs-changes

**Comment by adiroiban** [comment]
needs-review

**Comment by adiroiban** [comment]
changes-approved

**Comment by adiroiban** [comment]
changes-approved

**Comment by adiroiban** [comment]
changes-approved

**Comment by adiroiban** [comment]
changes-approved

**Comment by adiroiban** [comment]
changes-approved

**Comment by adiroiban** [comment]
changes-appproved

**Comment by adiroiban** [comment]
changes-approved

**Comment by adiroiban** [comment]
needs-review

**Comment by adiroiban** [comment]
needs-changes

**Comment by adiroiban** [comment]
needs-review

**Comment by adiroiban** [comment]
needs-review

**Comment by adiroiban** [comment]
needs-review

**Comment by hcs0** [comment]
changes-approved

**Comment by dumol** [comment]
Doesn't seem to be working as expected, at least not in this PR, as @hcs0 was not removed from the list of assignees after issuing a comment that approves the changes. Also, I've tried the _needs-changes_ label to no effect. By the way, I wonder: is the _needs-merge_ label equivalent to a comment for approving the changes?

It could be me though, as the PR's description is hard to comprehend, especially the _Changes_ part, which seems incomplete.

**needs-changes**

**Comment by hcs0** [comment]
@dumol 

When I first approved the PR ``chevah-robot`` unassigned me (see update ``@chevah-robot chevah-robot unassigned hcs0 20 hours ago``).  But later on I manually re-assigned myself and added a few labels to mess around with the PR a bit more.

**Comment by hcs0** [comment]
Interesting, looks like after @dumol added a comment that the PR needs changes, ``chevah-robot`` has unassigned all of us.  I was expecting that it will only unassign dumol while everyone else reviews the PR.

**Comment by adiroiban** [comment]
when it needs-changes, nobody will have to review it and the author needs to work on it, so this is why the author is the new assignee

**Comment by adiroiban** [comment]
up I have accidentally triggered the changes request :)

needs-review

**Comment by dumol** [comment]
PR description improved, things are more clear now. Let's see it working…

**changes-approved**

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,11 @@
+# git-secret protected file.
+chevah/github_hooks_server/tests/private.py
+
+.gitsecret/keys/pubring.gpg~
```
Comment (danuker) [inline .gitignore:4]: Test comment.

**Review by danuker — COMMENTED** [review]
Test review.

**Comment by danuker** [comment]
needs-changes

**Review by danuker — COMMENTED** [review]
Test review. changes-approved

**Comment by danuker** [comment]
needs-review

**Comment by danuker** [comment]
changes-approved

**Comment by danuker** [comment]
needs-review

**Review by danuker — COMMENTED** [review]
Testing review as a team.

**Review by danuker — COMMENTED** [review]
testing review as a team, while not the only one in the team

**Review by danuker — COMMENTED** [review]
testing review as a team, while not the only one in the team, pt 2.

**Comment by adiroiban** [comment]
test needs-changes

**Comment by adiroiban** [comment]
ok... just a test, please review

**Comment by danuker** [comment]
please review

Should get directed to the team

**Comment by danuker** [comment]
please review

Should get directed to the team

**Comment by danuker** [comment]
please review

again...

**Comment by danuker** [comment]
needs-review

**Comment by danuker** [comment]
changes-approved

**Comment by danuker** [comment]
needs-review

**Comment by danuker** [comment]
needs-changes

This should be ignored, as per the configuration.

File: .gitignore (hunk 1 above)
Comment (danuker) [reply .gitignore:4]: Another test comment.

**Review by danuker — COMMENTED** [review]
A test review changes-approved

**Pull request merged**